src/client/java/teammates/client/scripts/sql/DataMigrationForCourseSql.java

```diff
@@ -0,0 +1,57 @@
+package teammates.client.scripts.sql;
+
+import com.googlecode.objectify.cmd.Query;
+
+import teammates.common.util.HibernateUtil;
+import teammates.storage.entity.Course;
+
+/**
+ * Data migration class for course entity.
+ */
+@SuppressWarnings("PMD")
+public class DataMigrationForCourseSql extends
+        DataMigrationEntitiesBaseScriptSql<teammates.storage.entity.Course, teammates.storage.sqlentity.Course> {
+
+    public static void main(String[] args) {
+        new DataMigrationForCourseSql().doOperationRemotely();
+    }
+
+    @Override
+    protected Query<Course> getFilterQuery() {
+        return ofy().load().type(teammates.storage.entity.Course.class);
+    }
+
+    @Override
+    protected boolean isPreview() {
+        return false;
+    }
+
+    /*
+     * Sets the migration criteria used in isMigrationNeeded.
+     */
+    @Override
+    protected void setMigrationCriteria() {
+        // No migration criteria currently needed.
+    }
+
+    @Override
+    protected boolean isMigrationNeeded(Course entity) {
+        HibernateUtil.beginTransaction();
+        teammates.storage.sqlentity.Course course = HibernateUtil.get(
+                teammates.storage.sqlentity.Course.class, entity.getUniqueId());
+        HibernateUtil.commitTransaction();
+        return course == null;
+    }
```

Comment:
@NicolasCwy
Is it gonna be costly to query the sql db for every entity here?
Also do we need a `MarkIsMigrated` script for `Course` like the `MarkIsMigratedForAccounts.java` since Course has a `isMigrated` field? If needed then maybe can just check if `!isMigrated` ?

Reply:
For now we can comment the `isMigratedNeeded` body and return True @mingyuanc, let's get the scripts down then worry about improvements later

@yuanxi1 We'll see what the requirements are for migrating on prod and add this later if need be, for now you can just keep that script in mind if you need it later